docs/user-guide/ingress.md

```diff
@@ -0,0 +1,240 @@
+* Service: A Kubernetes [Service](https://github.com/kubernetes/kubernetes/blob/release-1.0/docs/user-guide/services.md) that identifies a set of pods using label selectors. Unless mentioned otherwise, Services are assumed to have virtual IPs only routable within the cluster network.
+
+## What is Ingress?
+
+Typically, services and pods have IPs only routable by the cluster network. All traffic that ends up at an edge router is either dropped or forwarded elsewhere. Conceptually, this might look like:
+
+```
+    internet
+        |
+  ------------
+  [ Services ]
+```
+
+An Ingress is a collection of rules that allow inbound connections to reach the cluster services.
+
+```
+    internet
+        |
+   [ Ingress ]
+   --|-----|--
+   [ Services ]
+```
+
+It can be configured to give services externally-reachable urls, load balance traffic, terminate SSL, offer name based virtual hosting etc. Users request ingress by POSTing the Ingress resource to the API server. An [Ingress controller](#ingress-controllers) is responsible for fulfilling the Ingress, usually with a loadbalancer, though it may also configure your edge router or additional frontends to help handle the traffic in an HA manner.
+
+## The Ingress Resource
+
+A minimal Ingress might look like:
+
+```yaml
+01. apiVersion: extensions/v1beta1
+02. kind: Ingress
+03. metadata:
+04.  name: hostlessendpoint
+05. spec:
+06.  rules:
+07.  - http:
+08.      paths:
+09.      - path: /hostless
+10.        backend:
+11.          serviceName: test
+12.          servicePort: 80
+```
+
+[Download example](ingress.yaml?raw=true)
+
+*POSTing this to the API server will have no effect if you have not configured an [Ingress controller](#ingress-controllers).*
+
+__Lines 1-4__: Resource metadata. An Ingress controller might use this to tag shared resources such as loadbalancers with a label or annotation.
+
+__Lines 5-7__: Ingress Spec has all the information needed to configure a loadbalancer or proxy server. Most importantly, it contains a list of rules matched against all incoming requests. Currently the Ingress resource only supports http rules.
+
+__Lines 8-9__: Each http rule contains the following information: A host (eg: foo.bar.com, defaults to * in this example), a list of paths (eg: /hostless) each of which has an associated backend (test:80). Both the host and path must match the content of an incoming request before the loadbalancer directs traffic to the backend.
+
+__Lines 10-12__: A backend is a service:port combination. It selects a group of pods capable of servicing traffic sent to the path specified in the parent rule.
```

Comment:
If you are trying to define how a kubernetes service works, here isn't the place to do it.  Use a link to services docs.  If you are saying something more subtle than that, then I missed it.